src/operator/nn/cudnn/cudnn_convolution-inl.h

```diff
@@ -224,6 +233,14 @@ class CuDNNConvolutionOp {
     CHECK_EQ(in_data.size(), expected);
     CHECK_EQ(in_grad.size(), expected);
     Stream<gpu> *s = ctx.get_stream<gpu>();
+    Stream<gpu> *s_dgrad = parallelize_backward_kernels_ ? ctx.get_aux_stream<gpu>() : s;
+
+    // Make sure the dgrad kernel in the aux stream doesn't start before it would have
+    // had it been launched into the operator's primary stream.
+    if (parallelize_backward_kernels_ && req[conv::kData] != kNullOp) {
+      CUDA_CALL(cudaEventRecord(dgrad_can_start_, s->stream_));
+      CUDA_CALL(cudaStreamWaitEvent(s_dgrad->stream_, dgrad_can_start_, 0));
+    }
```

Comment:
I'm a bit concerned on leaving such cross-stream coordination to the operators. I think better abstraction for this is needed.

Reply:
@ptrendx agrees and we've discussed an approach that would unburden operators from this low-level manipulation.  Hold off on a final review / merge of this PR until I've prototyped it.